src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/PackageItemViewModel.cs

```
@@ -494,14 +494,38 @@ public async Task<IReadOnlyCollection<VersionInfoContextInfo>> GetVersionsAsync(
         {
             var identity = new PackageIdentity(Id, Version);
             var isTransitive = PackageLevel == PackageLevel.Transitive;
-            return await _searchService.GetPackageVersionsAsync(identity, Sources, IncludePrerelease, isTransitive, _cancellationTokenSource.Token);
+
+            CancellationToken cancellationToken;
+            try
+            {
+                cancellationToken = _cancellationTokenSource.Token;
+            }
+            catch (ObjectDisposedException)
+            {
+                // If the CancellationTokenSource is disposed, return empty collection
+                return Array.Empty<VersionInfoContextInfo>();
```

Could this cause correctness issues? 
Wouldn't it be more accurate throw an exception and tear it down instead of return a result potentially indicating things are fine?